tests/ga/test_extension.py

```diff
@@ -1349,6 +1349,8 @@ def test_it_should_parse_in_vm_metadata_properly(self, mock_get, mock_crypt, *ar
         self.assertEqual(correlation_id, "400de90b-522e-491f-9d89-ec944661f531", "Incorrect correlation Id")
         self.assertEqual(gs_creation_time, '2020-11-09T17:48:50.412125Z', "Incorrect GS Creation time")
 
+    def test_it_should_parse_missing_in_vm_metadata_properly(self, mock_get, mock_crypt, *args):
```

Comment:
The name suggests that the function would parse metadata even if its missing. Maybe rename to `test_it_should_process_goal_state_even_if_metadata_missing` and merge the two tests in one?

Reply:
I would like to keep the tests separate because they're all using different data files, but I agree about the rename